WalletWasabi/Crypto/GroupElement.cs

```diff
@@ -100,6 +101,44 @@ public override string ToString()
 			return a + new GroupElement(b.Ge.Negate());
 		}
 
+		/// <param name="scalar">It's ok for the scalar to overflow.</param>
+		public static GroupElement operator *(Scalar scalar, GroupElement groupElement)
+		{
+			Guard.NotNull(nameof(GroupElement), groupElement);
+
+			// For some strange reason scalar * GE.Infinity isn't infinity. Let's fix it as it should be, since:
+			// 2 * GE.Infinity = GE.Infinity + GE.Infinity = GE.Infinity.
+			if (groupElement.IsInfinity)
+			{
+				return Infinity;
+			}
+
+			return new GroupElement(scalar * groupElement.Ge);
+		}
+
+		/// <param name="scalar">It's ok for the scalar to overflow.</param>
+		public static GroupElement operator *(GroupElement groupElement, Scalar scalar) => scalar * groupElement;
+
 		public GroupElement Negate() => new GroupElement(Ge.Negate());
+
+		public byte[] ToBytes() => ByteHelpers.Combine(Ge.x.ToBytes(), Ge.y.ToBytes());
```

Comment:
I was planning to use it here: https://github.com/zkSNACKs/WalletWasabi/pull/3915/files#diff-94391fcf9b0bdadb1f101ea58c077b6eR20-R24

Is using the compressed there ok?

Reply:
Are we sure this makes sense though? We're doing gzip compression when sending data over the network by default. We encountered before that the custom compression didn't overperform our default compression, so I'm not sure we want to introduce this complexity for no reason.

Reply:
for the proofs of knowledge it doesn't matter which encoding is used so long as it's always the same, so i don't see a problem with uncompressed encoding there.

on the wire, gzip compression probably won't help since the coordinate values look random unless related by the curve equation.